tuxemon/condition/condeffect.py

```diff
@@ -14,11 +14,13 @@
     from tuxemon.technique.technique import Technique
 
 
-class CondEffectResult(TypedDict):
+@dataclass
+class CondEffectResult:
+    name: str
     success: bool
-    condition: Union[Condition, None]
-    technique: Union[Technique, None]
-    extra: Union[str, None]
+    condition: list[Condition]
```

Comment:
As these are now lists you should rename to `conditions`, `techniques`, `extras`.